pkg/shares/split_compact_shares.go

```diff
@@ -57,10 +57,11 @@ func (css *CompactShareSplitter) WriteEvidence(evd coretypes.Evidence) error {
 
 // WriteBytes adds the delimited data to the underlying compact shares.
 func (css *CompactShareSplitter) WriteBytes(rawData []byte) {
-	// if this is the first time writing to a pending share, we must add the
-	// reserved bytes
+	// if this is the first time writing to a pending share, we must add a
+	// placeholder for the reserved bytes
```

Comment:
[question]
It's been a minute since I've looked at this code, is this a placeholder that we overwrite or is this an edge case where we actually want zeros?

Reply:
it seems like a placeholder but it doesn't look like it's overwritten for [this case](https://github.com/celestiaorg/celestia-app/pull/801/files#diff-3da7e413fef3c90c01286110ba9bdfc3d61bbb360355bb32adede4e7b3ebfe5bR74-R77) so we need unit tests to identify if there's a bug there

Reply:
agree, I don't think we have an explicit test for reserved bytes and we just ignore them when merging.